### PR TITLE
Gainsight NXT new components

### LINKED ISSUE #14373: [Components] gainsight_nxt
# gainsight_nxt
## URLs
- https://support.gainsight.com/gainsight_nxt/API_and_Developer_Docs
#### Prompt
Create or update a company record. Users need to configure the 'All Fields' props with the relevant company information. If the company exists, this action updates the information; if not, it creates a new record.
#### Prompt
Create or modify a custom object. Configurable via 'All Fields' props, the user can specify the custom object elements. If it exists, the data gets updated; otherwise, a new custom object is created.
#### Prompt
Add or update a person's record. 'All Fields' props allow dynamic input of a person's data. In case of an existing profile, the data is updated. If not, a new person's profile is generated.

### PR DESCRIPTION
**Pull request by GTFalcao**
Closes #14373 

## Release Notes

- **New Features**
  - Introduced modules for creating or updating company, custom object, and person records in Gainsight NXT.
  - Enhanced API interaction capabilities with new methods for managing company and custom object data.

- **Bug Fixes**
  - Improved error handling for create/update operations to ensure seamless transitions.

- **Documentation**
  - Updated package version to 0.1.0 and added dependencies for better functionality.

- **Chores**
  - Added utility functions for improved JSON parsing and object entry transformation.

### CODE REVIEW

File: components/gainsight_nxt/gainsight_nxt.app.mjs [hunk 1]
```diff
@@ -1,11 +1,92 @@
...
   app: "gainsight_nxt",
-  propDefinitions: {},
+  propDefinitions: {
+    objectName: {
+      type: "string",
+      label: "Custom Object",
+      description: "The name of the custom object.",
+      async options() {
+        const { data } = await this.listCustomObjects();
+        return data/*?.filter?.((obj) => obj.objectType === "CUSTOM")*/.map(( {
```
Comment (michelle0927) [inline components/gainsight_nxt/gainsight_nxt.app.mjs:13]: Is the filter meant to be commented out?

**Review by michelle0927 — CHANGES_REQUESTED** [review]

File: components/gainsight_nxt/gainsight_nxt.app.mjs (hunk 1 above)
Comment (GTFalcao) [reply components/gainsight_nxt/gainsight_nxt.app.mjs:13]: It is, I commented it out for testing and forgot to remove the block afterwards - doing it now, thanks.

For the record, it seems several custom object types are not actually classified explicitly as "custom", hence why I removed this filter (it was returning no objects whatsoever with it)

**Review by michelle0927 — DISMISSED** [review]
LGTM!

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**